internal/models/models.go

```diff
@@ -0,0 +1,28 @@
+// The models package defines internal structs that uses for validate/generate proof.
+
+package models
+
+import (
+	"math/big"
+
+	bn256 "github.com/ethereum/go-ethereum/crypto/bn256/cloudflare"
+)
+
+// R is the mod of the finite field
+var R, _ = new(big.Int).SetString("21888242871839275222246405745257275088548364400416034343698204186575808495617", 10)
+
+// ProofData describes three components of zkp proof in bn256 format.
+type ProofData struct {
```

Comment:
it's better to rename it to ProofPairingData and why do we need internal/models ?